launchable/utils/session.py

```diff
@@ -0,0 +1,70 @@
+import click
+import os
+import json
+from pathlib import Path
+from typing import Optional
+
+session_file_path = Path('~/.config/launchable/session.json').expanduser()
+
+
+class SessionError(Exception):
+    pass
+
+
+def read_session(build_name: str) -> Optional[str]:
+    if not session_file_path.exists():
+        raise SessionError(
+            "No session file. Please write session {}".format(session_file_path))
+
+    with session_file_path.open('r') as json_file:
+        try:
+            data = json.load(json_file)
+            return data.get("{}:{}".format(build_name, os.getsid(os.getpid())))
+        except json.JSONDecodeError as e:
+            raise SessionError(
+                "Invalid session file. Please remove {}".format(session_file_path))
+
+
+def write_session(build_name: str, session_id: str) -> None:
+    if session_file_path.exists():
+        with session_file_path.open('r') as json_file:
+            try:
+                data = json.load(json_file)
+            except json.JSONDecodeError as e:
+                raise SessionError(
+                    "Invalid session file. Please remove {}".format(session_file_path))
+    else:
+        session_file_path.parent.mkdir(parents=True, exist_ok=True)
+        session_file_path.touch()
+        data = {}
+
+    with session_file_path.open('w') as json_file:
+        data["{}:{}".format(build_name, os.getsid(os.getpid()))] = session_id
+        json.dump(data, json_file)
+
+
+def remove_session(build_name: str) -> None:
+    """
+    Call it after closing a session
+    """
+    if not session_file_path.exists():
+        raise SessionError(
+            "No session file. Please write session {}".format(session_file_path))
+    with session_file_path.open('r') as json_file:
+        try:
+            data = json.load(json_file)
+        except json.JSONDecodeError as e:
+            raise SessionError(
+                "Invalid session file. Please remove {}".format(session_file_path))
```

Comment:
This also feels like we'd be better off just clobbering this file altogether. Discussed this more extensively above.